Sources/Update/Checkout.swift

```diff
@@ -0,0 +1,43 @@
+import struct PackageDescription.Version
+import PackageType
+import Utility
+import Get
+
+struct Checkout: Equatable, Hashable {
```

Comment:
It seems to me like `Checkout` should be a class and just use instance equality? Two checkouts with the same URL but different repos or manifests aren't really equivalent.